build-logic/jvm/src/main/kotlin/build-logic.jacoco-conventions.gradle.kts

```diff
@@ -0,0 +1,36 @@
+import org.gradle.api.Project
+import org.gradle.kotlin.dsl.extra
+import org.gradle.kotlin.dsl.findByType
+import org.gradle.testing.jacoco.plugins.JacocoPluginExtension
+import org.gradle.testing.jacoco.tasks.JacocoReport
+
+val jacocoToolVersion: String by rootProject.extra
+
+plugins.withId("jacoco") {
+    configure<JacocoPluginExtension> {
+        if (rootProject.name != "gradle-kotlin-dsl-accessors") {
+            toolVersion = jacocoToolVersion
+        }
+    }
+
+    val jacocoAdditionalExtraName = "jacocoAdditional"
```

Comment:
This should be replaced with a proper plugin, so it exposes a regular extension instead of relying on `project.extra`

Reply:
feel free to improve it 👍